polars/polars-arrow/src/data_types.rs

```diff
@@ -20,6 +20,8 @@ impl IsFloat for u8 {}
 impl IsFloat for u16 {}
 impl IsFloat for u32 {}
 impl IsFloat for u64 {}
+impl IsFloat for &str {}
+impl<T: IsFloat> IsFloat for Option<T> {}
```

Comment:
This implementation is dangerous, as users could implement `PartialOrd for Option<LocalType>` in ways that disagree with this.

Reply:
I will make it an unsafe trait. For all types this is intended to be used for this is correct.